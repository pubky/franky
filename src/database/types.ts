// Primary keys for database entities
export type UserPK = string; // User's public key
export type PostPK = `${UserPK}:${string}`; // Format: userPK:postIdentifier

// Timestamp type used across the database
export type Timestamp = number;

// Sync status for database entities
export type SyncStatus = 'local' | 'homeserver' | 'nexus';

export type HomeserverActions = 'PUT' | 'DEL';
