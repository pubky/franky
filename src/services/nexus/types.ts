import { type UserPK, type PostPK, type Timestamp } from '@/database/types';

// Common types

export type NexusPostKind = 'short' | 'long' | 'repost' | 'reply' | 'link';

export interface NexusBookmark {
  created_at: number;
  updated_at: number;
}

export interface NexusUserLink {
  title: string;
  url: string;
}

// User types
export interface NexusUserDetails {
  name: string;
  bio: string;
  id: UserPK;
  links: NexusUserLink[] | null;
  status: string | null;
  image: string | null;
  indexed_at: Timestamp;
}

export interface NexusUserCounts {
  tagged: number; // the number of tags assign by this user to other entities ('users' or 'posts')
  tags: number; // the number of tags received by this user from other users
  unique_tags: number; // the number of unique tags received by this user from other users (distinct)
  posts: number;
  replies: number;
  following: number;
  followers: number;
  friends: number;
  bookmarks: number;
}

export interface NexusUserRelationship {
  following: boolean;
  followed_by: boolean;
  muted: boolean;
}

export interface NexusUserTag {
  label: string;
  taggers: UserPK[];
  taggers_count: number;
  relationship: boolean;
}

export interface NexusUser {
  details: NexusUserDetails;
  counts: NexusUserCounts;
  tags: NexusUserTag[];
  relationship: NexusUserRelationship;
}

// Post types
export interface NexusPostDetails {
  content: string;
  id: PostPK;
  indexed_at: number;
  author: UserPK;
  kind: NexusPostKind;
  uri: string;
  attachments: string[] | null;
}

export interface NexusPostCounts {
  tags: number;
  unique_tags: number;
  replies: number;
  reposts: number;
}

export interface NexusPostTag {
  label: string;
  taggers: UserPK[];
  taggers_count: number;
  relationship: boolean;
}

export interface NexusPostRelationships {
  replied: string | null;
  reposted: string | null;
  mentioned: UserPK[];
}

export interface NexusPost {
  details: NexusPostDetails;
  counts: NexusPostCounts;
  tags: NexusPostTag[];
  relationships: NexusPostRelationships;
  bookmark: NexusBookmark | null;
}

export interface NexusList {
  stream: string[];
  influencers: UserPK[];
  recommended: UserPK[];
}

export interface NexusBootstrapResponse {
  users: NexusUser[];
  posts: NexusPost[];
  list: NexusList;
}
