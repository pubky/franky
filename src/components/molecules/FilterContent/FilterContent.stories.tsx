import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import { useState } from 'react';

import { FilterContent, type ContentTab } from './FilterContent';

const meta: Meta<typeof FilterContent> = {
  title: 'Molecules/Filter/FilterContent',
  component: FilterContent,
  parameters: {
    layout: 'centered',
    docs: {
      description: {
        component: 'Preset content filter with iconography built from the filter atom primitives.',
      },
    },
  },
  args: {
    selectedTab: 'all',
  },
  argTypes: {
    selectedTab: {
      control: 'select',
      options: ['all', 'posts', 'articles', 'images', 'videos', 'links', 'files'],
    },
    onTabChange: { action: 'tab changed' },
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Playground: Story = {
  render: (args) => <FilterContent {...args} />,
  parameters: {
    docs: {
      description: {
        story: 'Adjust the controls to simulate different selected tabs through args.',
      },
    },
  },
};

export const Interactive: Story = {
  render: () => {
    const [selectedTab, setSelectedTab] = useState<ContentTab>('all');

    return (
      <FilterContent
        selectedTab={selectedTab}
        onTabChange={(tab) => {
          setSelectedTab(tab);
        }}
      />
    );
  },
  parameters: {
    docs: {
      description: {
        story: 'Internal state example showing how user interaction changes the selected content filter.',
      },
    },
  },
};

export const VideosSelected: Story = {
  args: {
    selectedTab: 'videos',
  },
};

export const LinksSelected: Story = {
  args: {
    selectedTab: 'links',
  },
};
