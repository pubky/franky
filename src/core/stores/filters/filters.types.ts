// Filter constants
export const LAYOUT = {
  COLUMNS: 'columns',
  WIDE: 'wide',
  VISUAL: 'visual',
} as const;

export const SORT = {
  RECENT: 'recent',
  POPULARITY: 'popularity',
} as const;

export const REACH = {
  ALL: 'all',
  FOLLOWING: 'following',
  FRIENDS: 'friends',
  ME: 'me',
} as const;

export const CONTENT = {
  ALL: 'all',
  POSTS: 'posts',
  ARTICLES: 'articles',
  IMAGES: 'images',
  VIDEOS: 'videos',
  LINKS: 'links',
  FILES: 'files',
} as const;

// Filter types
export type LayoutType = (typeof LAYOUT)[keyof typeof LAYOUT];
export type SortType = (typeof SORT)[keyof typeof SORT];
export type ReachType = (typeof REACH)[keyof typeof REACH];
export type ContentType = (typeof CONTENT)[keyof typeof CONTENT];

export interface FiltersState {
  layout: LayoutType;
  sort: SortType;
  reach: ReachType;
  content: ContentType;
}

export interface FiltersActions {
  setLayout: (layout: LayoutType) => void;
  setSort: (sort: SortType) => void;
  setReach: (reach: ReachType) => void;
  setContent: (content: ContentType) => void;
  reset: () => void;
}

export type FiltersStore = FiltersState & FiltersActions;

// Initial state
export const filtersInitialState: FiltersState = {
  layout: LAYOUT.COLUMNS,
  sort: SORT.RECENT,
  reach: REACH.ALL,
  content: CONTENT.ALL,
};

// Action types for DevTools
export enum FiltersActionTypes {
  SET_LAYOUT = 'filters/setLayout',
  SET_SORT = 'filters/setSort',
  SET_REACH = 'filters/setReach',
  SET_CONTENT = 'filters/setContent',
  RESET = 'filters/reset',
}
