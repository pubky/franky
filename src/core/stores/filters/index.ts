export * from './filters.store';
export * from './filters.types';
export * from './filters.actions';
