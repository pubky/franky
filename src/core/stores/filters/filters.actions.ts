import { FiltersStore, FiltersActions, filtersInitialState, FiltersActionTypes } from './filters.types';
import { ZustandSet } from '../stores.types';

// Actions/Mutators - State modification functions
export const createFiltersActions = (set: ZustandSet<FiltersStore>): FiltersActions => ({
  setLayout: (layout) => {
    set({ layout }, false, FiltersActionTypes.SET_LAYOUT);
  },

  setSort: (sort) => {
    set({ sort }, false, FiltersActionTypes.SET_SORT);
  },

  setReach: (reach) => {
    set({ reach }, false, FiltersActionTypes.SET_REACH);
  },

  setContent: (content) => {
    set({ content }, false, FiltersActionTypes.SET_CONTENT);
  },

  reset: () => {
    set(filtersInitialState, false, FiltersActionTypes.RESET);
  },
});
