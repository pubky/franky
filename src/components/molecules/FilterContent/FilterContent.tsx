'use client';

import * as Atoms from '@/atoms';
import * as Libs from '@/libs';
import { CONTENT, type ContentType } from '@/core/stores/filters/filters.types';

export type ContentTab = ContentType;

interface ContentProps {
  selectedTab?: ContentTab;
  onTabChange?: (tab: ContentTab) => void;
}

const tabs: { key: ContentTab; label: string; icon: React.ComponentType<{ className?: string }> }[] = [
  { key: CONTENT.ALL, label: 'All', icon: Libs.Layers },
  { key: CONTENT.POSTS, label: 'Posts', icon: Libs.StickyNote },
  { key: CONTENT.ARTICLES, label: 'Articles', icon: Libs.Newspaper },
  { key: CONTENT.IMAGES, label: 'Images', icon: Libs.Image },
  { key: CONTENT.VIDEOS, label: 'Videos', icon: Libs.CirclePlay },
  { key: CONTENT.LINKS, label: 'Links', icon: Libs.Link },
  { key: CONTENT.FILES, label: 'Files', icon: Libs.Download },
];

export function FilterContent({ selectedTab = CONTENT.ALL, onTabChange }: ContentProps) {
  const handleTabClick = (tab: ContentTab) => {
    onTabChange?.(tab);
  };

  return (
    <Atoms.FilterRoot>
      <Atoms.FilterHeader title="Content" />

      <Atoms.FilterList>
        {tabs.map(({ key, label, icon: Icon }) => {
          const isSelected = selectedTab === key;

          return (
            <Atoms.FilterItem key={key} isSelected={isSelected} onClick={() => handleTabClick(key)}>
              <Atoms.FilterItemIcon icon={Icon} />
              <Atoms.FilterItemLabel>{label}</Atoms.FilterItemLabel>
            </Atoms.FilterItem>
          );
        })}
      </Atoms.FilterList>
    </Atoms.FilterRoot>
  );
}
