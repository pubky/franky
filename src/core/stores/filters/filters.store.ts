import { create } from 'zustand';
import { persist, devtools } from 'zustand/middleware';
import { FiltersStore, filtersInitialState } from './filters.types';
import { createFiltersActions } from './filters.actions';

// Store creation
export const useFiltersStore = create<FiltersStore>()(
  devtools(
    persist(
      (set) => ({
        ...filtersInitialState,
        ...createFiltersActions(set),
      }),
      {
        name: 'filters-store',
        // Persist all filter states
        partialize: (state) => ({
          layout: state.layout,
          sort: state.sort,
          reach: state.reach,
          content: state.content,
        }),
      },
    ),
    {
      name: 'filters-store',
      enabled: process.env.NODE_ENV === 'development',
    },
  ),
);
