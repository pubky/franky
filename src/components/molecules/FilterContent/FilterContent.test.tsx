import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { FilterContent, type ContentTab } from './FilterContent';
import { CONTENT } from '@/core/stores/filters/filters.types';

// Mock libs - use actual utility functions and icons from lucide-react
vi.mock('@/libs', async (importOriginal) => {
  const actual = await importOriginal<typeof import('@/libs')>();
  return { ...actual };
});

describe('FilterContent', () => {
  it('renders with default selected tab', () => {
    render(<FilterContent />);

    expect(screen.getByText('Content')).toBeInTheDocument();
  });

  it('calls onTabChange when tab is clicked', () => {
    const onTabChange = vi.fn();
    render(<FilterContent onTabChange={onTabChange} />);

    fireEvent.click(screen.getByText('Images'));
    expect(onTabChange).toHaveBeenCalledWith('images');
  });

  it('handles all tab types correctly', () => {
    const onTabChange = vi.fn();
    render(<FilterContent onTabChange={onTabChange} />);

    const tabs: ContentTab[] = ['all', 'posts', 'articles', 'images', 'videos', 'links', 'files'];

    tabs.forEach((tab) => {
      // capitalize first letter
      const label = tab.charAt(0).toUpperCase() + tab.slice(1);
      fireEvent.click(screen.getByText(label));
      expect(onTabChange).toHaveBeenCalledWith(tab);
    });
  });

  it('applies correct styling classes', () => {
    render(<FilterContent selectedTab={CONTENT.ALL} />);

    const allItem = screen.getByText('All').closest('[data-testid="filter-item"]');
    const postsItem = screen.getByText('Posts').closest('[data-testid="filter-item"]');

    expect(allItem).toMatchSnapshot();
    expect(postsItem).toMatchSnapshot();
  });

  it('renders with correct icons', () => {
    render(<FilterContent />);

    expect(screen.getByTestId('filter-list')).toMatchSnapshot();
  });

  it('applies correct icon classes', () => {
    render(<FilterContent />);

    expect(screen.getByTestId('filter-list')).toMatchSnapshot();
  });

  it('handles tab switching correctly', () => {
    const onTabChange = vi.fn();
    render(<FilterContent selectedTab={CONTENT.ALL} onTabChange={onTabChange} />);

    // Click on articles tab
    fireEvent.click(screen.getByText('Articles'));
    expect(onTabChange).toHaveBeenCalledWith('articles');
    expect(onTabChange).toHaveBeenCalledTimes(1);
  });

  it('rerenders with different selected tabs', () => {
    const { rerender } = render(<FilterContent selectedTab="all" />);

    let allItem = screen.getByText('All').closest('[data-testid="filter-item"]');
    const postsItem = screen.getByText('Posts').closest('[data-testid="filter-item"]');
    expect(allItem).toMatchSnapshot();
    expect(postsItem).toMatchSnapshot();

    // Rerender with different selected tab
    rerender(<FilterContent selectedTab="videos" />);
    allItem = screen.getByText('All').closest('[data-testid="filter-item"]');
    const videosItem2 = screen.getByText('Videos').closest('[data-testid="filter-item"]');
    expect(allItem).toMatchSnapshot();
    expect(videosItem2).toMatchSnapshot();
  });

  it('handles multiple tab clicks', () => {
    const onTabChange = vi.fn();
    render(<FilterContent onTabChange={onTabChange} />);

    fireEvent.click(screen.getByText('Images'));
    fireEvent.click(screen.getByText('Videos'));
    fireEvent.click(screen.getByText('Files'));

    expect(onTabChange).toHaveBeenCalledTimes(3);
    expect(onTabChange).toHaveBeenNthCalledWith(1, 'images');
    expect(onTabChange).toHaveBeenNthCalledWith(2, 'videos');
    expect(onTabChange).toHaveBeenNthCalledWith(3, 'files');
  });
});

describe('FilterContent - Snapshots', () => {
  it('matches snapshot with default props', () => {
    const { container } = render(<FilterContent />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with All content selected tab', () => {
    const { container } = render(<FilterContent selectedTab={CONTENT.ALL} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Posts content selected tab', () => {
    const { container } = render(<FilterContent selectedTab={CONTENT.POSTS} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Articles content selected tab', () => {
    const { container } = render(<FilterContent selectedTab={CONTENT.ARTICLES} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Images content selected tab', () => {
    const { container } = render(<FilterContent selectedTab={CONTENT.IMAGES} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Videos content selected tab', () => {
    const { container } = render(<FilterContent selectedTab={CONTENT.VIDEOS} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Links content selected tab', () => {
    const { container } = render(<FilterContent selectedTab={CONTENT.LINKS} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Files content selected tab', () => {
    const { container } = render(<FilterContent selectedTab={CONTENT.FILES} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
