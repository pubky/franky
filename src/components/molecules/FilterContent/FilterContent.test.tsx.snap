// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterContent > applies correct icon classes 1`] = `
<div
  class="mx-auto w-full flex-col flex flex flex-col gap-2"
  data-slot="filter-list"
  data-testid="filter-list"
>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
    data-selected="true"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="layers-icon"
    >
      Layers
    </div>
    <span
      class=""
    >
      All
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="sticky-note-icon"
    >
      StickyNote
    </div>
    <span
      class=""
    >
      Posts
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="newspaper-icon"
    >
      Newspaper
    </div>
    <span
      class=""
    >
      Articles
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="image-icon"
    >
      Image
    </div>
    <span
      class=""
    >
      Images
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="circle-play-icon"
    >
      CirclePlay
    </div>
    <span
      class=""
    >
      Videos
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="link-icon"
    >
      Link
    </div>
    <span
      class=""
    >
      Links
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="download-icon"
    >
      Download
    </div>
    <span
      class=""
    >
      Files
    </span>
  </p>
</div>
`;

exports[`FilterContent > applies correct styling classes 1`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="layers-icon"
  >
    Layers
  </div>
  <span
    class=""
  >
    All
  </span>
</p>
`;

exports[`FilterContent > applies correct styling classes 2`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="sticky-note-icon"
  >
    StickyNote
  </div>
  <span
    class=""
  >
    Posts
  </span>
</p>
`;

exports[`FilterContent > renders with correct icons 1`] = `
<div
  class="mx-auto w-full flex-col flex flex flex-col gap-2"
  data-slot="filter-list"
  data-testid="filter-list"
>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
    data-selected="true"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="layers-icon"
    >
      Layers
    </div>
    <span
      class=""
    >
      All
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="sticky-note-icon"
    >
      StickyNote
    </div>
    <span
      class=""
    >
      Posts
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="newspaper-icon"
    >
      Newspaper
    </div>
    <span
      class=""
    >
      Articles
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="image-icon"
    >
      Image
    </div>
    <span
      class=""
    >
      Images
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="circle-play-icon"
    >
      CirclePlay
    </div>
    <span
      class=""
    >
      Videos
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="link-icon"
    >
      Link
    </div>
    <span
      class=""
    >
      Links
    </span>
  </p>
  <p
    class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <div
      class="w-5 h-5"
      data-testid="download-icon"
    >
      Download
    </div>
    <span
      class=""
    >
      Files
    </span>
  </p>
</div>
`;

exports[`FilterContent > renders with custom selected tab 1`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="circle-play-icon"
  >
    CirclePlay
  </div>
  <span
    class=""
  >
    Videos
  </span>
</p>
`;

exports[`FilterContent > renders with default selected tab 1`] = `
<div
  class="mx-auto w-full flex-col flex flex flex-col gap-4 bg-background"
  data-slot="filter-root"
  data-testid="filter-root"
>
  <div
    data-slot="filter-header"
    data-testid="filter-header"
  >
    <h2
      class="text-2xl font-bold text-foreground text-muted-foreground font-light"
      data-testid="heading-2"
    >
      Content
    </h2>
  </div>
  <div
    class="mx-auto w-full flex-col flex flex flex-col gap-2"
    data-slot="filter-list"
    data-testid="filter-list"
  >
    <p
      class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
      data-selected="true"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <div
        class="w-5 h-5"
        data-testid="layers-icon"
      >
        Layers
      </div>
      <span
        class=""
      >
        All
      </span>
    </p>
    <p
      class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
      data-selected="false"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <div
        class="w-5 h-5"
        data-testid="sticky-note-icon"
      >
        StickyNote
      </div>
      <span
        class=""
      >
        Posts
      </span>
    </p>
    <p
      class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
      data-selected="false"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <div
        class="w-5 h-5"
        data-testid="newspaper-icon"
      >
        Newspaper
      </div>
      <span
        class=""
      >
        Articles
      </span>
    </p>
    <p
      class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
      data-selected="false"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <div
        class="w-5 h-5"
        data-testid="image-icon"
      >
        Image
      </div>
      <span
        class=""
      >
        Images
      </span>
    </p>
    <p
      class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
      data-selected="false"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <div
        class="w-5 h-5"
        data-testid="circle-play-icon"
      >
        CirclePlay
      </div>
      <span
        class=""
      >
        Videos
      </span>
    </p>
    <p
      class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
      data-selected="false"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <div
        class="w-5 h-5"
        data-testid="link-icon"
      >
        Link
      </div>
      <span
        class=""
      >
        Links
      </span>
    </p>
    <p
      class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
      data-selected="false"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <div
        class="w-5 h-5"
        data-testid="download-icon"
      >
        Download
      </div>
      <span
        class=""
      >
        Files
      </span>
    </p>
  </div>
</div>
`;

exports[`FilterContent > rerenders with different selected tabs 1`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="layers-icon"
  >
    Layers
  </div>
  <span
    class=""
  >
    All
  </span>
</p>
`;

exports[`FilterContent > rerenders with different selected tabs 2`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="sticky-note-icon"
  >
    StickyNote
  </div>
  <span
    class=""
  >
    Posts
  </span>
</p>
`;

exports[`FilterContent > rerenders with different selected tabs 3`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="layers-icon"
  >
    Layers
  </div>
  <span
    class=""
  >
    All
  </span>
</p>
`;

exports[`FilterContent > rerenders with different selected tabs 4`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="circle-play-icon"
  >
    CirclePlay
  </div>
  <span
    class=""
  >
    Videos
  </span>
</p>
`;

exports[`FilterContent > shows correct visual state for selected and unselected tabs 1`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="sticky-note-icon"
  >
    StickyNote
  </div>
  <span
    class=""
  >
    Posts
  </span>
</p>
`;

exports[`FilterContent > shows correct visual state for selected and unselected tabs 2`] = `
<p
  class="text-sm font-semibold text-foreground cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <div
    class="w-5 h-5"
    data-testid="layers-icon"
  >
    Layers
  </div>
  <span
    class=""
  >
    All
  </span>
</p>
`;
